add circle_fit to poetry.lock

hgloeckner opened the pull request.

hgloeckner commented:
closes #80

ninarobbins commented:
Hey @hgloeckner, thank you for this nice pull request! I keep getting the error `TypeError: Failed to decode variable 'launch_time': cannot inherit frozen dataclass from a non-frozen one` when running the pipeline. Do you know if there is something special I should do regarding poetry?

hgloeckner commented:
Hm. have you renewed your python environment? just a complete reset with conda?

tmieslinger reviewed (commented):
Thanks for this major improvement on the thermodynamic quantities!
The dependencies work fine for me and the change in some moist thermo variables seems negligible (for ta, theta), and somewhat visible for theta_e (0.5% lower) and remarkable (8% lower) for RH on average for the one random test case that I ran for all sondes on 2024-09-14. I think we should merge the PR and continue in a seperate thread adding tests to the moist thermodynamics repository to check consistency between forward and backward calculations etc.

ninarobbins commented on pydropsonde/helper/physics.py:
```diff
@@ -1,93 +1,70 @@
-        * (53.878 - 1331.22 / T - 9.44523 * np.log(T) + 0.014025 * T)
-    )
+def theta2ta(theta, P, qv=0.0, ql=0.0, qi=0.0):
+    """Returns the temperature for an unsaturated moist fluid, given the temperature
```
given potential temperature?

ninarobbins commented on pydropsonde/helper/__init__.py:
```diff
@@ -282,72 +286,30 @@ def get_si_converter_function_based_on_var(var_name):
-        g[5] = 7.0229056 * 10 ** (-10)
-        g[6] = -1.8680009 * 10 ** (-13)
-        g[7] = 2.7150305
+def q2mr(q):
```
very minor but maybe have the description here as in the other functions?

ninarobbins reviewed (commented):
It works well for me! I guess very minor comment but  some functions don't have a description yet, but this can also be done later.

hgloeckner commented:
I added a description to those functions and moved them to the physics file as they fit there better^^' 

> It works well for me! I guess very minor comment but some functions don't have a description yet, but this can also be done later.

hgloeckner commented:
> Thanks for this major improvement on the thermodynamic quantities! The dependencies work fine for me and the change in some moist thermo variables seems negligible (for ta, theta), and somewhat visible for theta_e (0.5% lower) and remarkable (8% lower) for RH on average for the one random test case that I ran for all sondes on 2024-09-14. I think we should merge the PR and continue in a seperate thread adding tests to the moist thermodynamics repository to check consistency between forward and backward calculations etc.

thank you so much for checking! I think, now it's also fairly easy to change the method for RH/theta calculation
